Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -134,9 +189,16 @@ public Task CloseAsync(CancellationToken cancellationToken)
             }
             else
             {
-                if ((_connectionState != ConnectionState.Closed) && SfSession != null)
+                if (IsNonClosedWithSession())
                 {
-                    if (SnowflakeDbConnectionPool.addSession(SfSession))
+                    var transactionInProgress = GetTransactionId();

```

Comment:
isn't it duplicated code with this one some lines above? are we able to extract method?

Reply:
There's a plan to have bigger changes in the connection pool. That's why I just wanted to do minimal changes here